### DISCUSSION TITLE
SPI TFT Screen support | Beginner friendly reface dx like mode

### DISCUSSION THREAD

**Discussion by jtomes123** [probonopd/MiniDexed#418] [Ideas]
Hi guys, first off I love the project, I set it up during the weekend and instantly ordered parts for a permanent build. 

Was there a discussion about supporting some of the SPI TFT screens?  There is a driver for the ST7789 in circle, however it would require some custom code to get it working with the current implementation. If there would be interest, I can take a look at that. I have some experience with HDMI using circle on the pi zero. You can get these screens in 2.8 inch variants and 320x240 resolution for couple dollars from aliexpress, so it would be a nice cheap way to have a bigger graphical display.

We could then easily draw the algorithms, envelopes, etc…

One more idea I had was to make a configuration for dexed and have it emulate the reface dx, but I have limited knowledge of dexed codebase, so I have no idea if it would be possible.

**Comment by diyelectromusic** [comment]
That would be really neat, but there is problem - the ST7789 support in circle is as a "graphical" device, but MiniDexed uses the "character device" support for the menus/etc.  The SSD1306 support in circle was put in (by me) as a character device to allow it to be used instead of a HD44780 - it still doesn't have full graphical support either.

So you'd either need a way to shim a character device onto the ST7789 display in circle itself (which would be my preferred method tbh), which can then be used by MiniDexed, or completely re-write the entire MiniDexed UI as a graphical, not character, UI (not keen on this, but arguably this is what would be required to do things like draw waveforms, etc).  In the latter case, we'd need a way to fall back on character only UI for the HD44780...

The display support in circle could really do with a revamp to properly support graphical displays, but that is probably a little way down the "to do" list for Rene I'd imagine.

Kevin

**Reply by jtomes123** [reply]
I was thinking of shiming in the support for current ui and slowly building up new graphical ui after that, the current UI would then also act as a fallback.

MicroDexed and reface dx could both act as a good inspiration for UI, however it is probebly best to be careful around any current-ish gen yamaha stuff because of copyright and patents.
